pyuvdata/uvdata/uvh5.py

```diff
@@ -1214,6 +1294,19 @@ def _write_header(self, header):
             header["eq_coeffs_convention"] = np.string_(self.eq_coeffs_convention)
         if self.flex_spw_id_array is not None:
             header["flex_spw_id_array"] = self.flex_spw_id_array
+        if self.phase_center_id_array is not None:
+            header["phase_center_id_array"] = self.phase_center_id_array
+        if self.Nphase is not None:
+            header["Nphase"] = self.Nphase
+
+        # Write out the catalog, if available
+        if self.phase_center_catalog:
+            phase_dict = header.create_group("phase_center_catalog")
+            for k in self.phase_center_catalog.keys():
+                # Dictionary entries can be written out as strings, in what is
```

Comment:
Thanks for updating how this is done! Just curious, what is the "str promotion" the comment is referring to? Also, my guess for what the datatype of the `phase_dict[k]` dataset would be is `numpy.object_`, which is usually what happens when you write a `str` dataset instead of first casting with `np.string_` or `np.bytes_` (which I believe are equivalent to each other). Should we be wrapping the call to `json.dumps` with a cast to `np.string_`? Or does that truncate the dictionary like the comment claims?

Reply:
Whoops -- that comment is actually out of date, relating back to when I was storing the entry as a "normal" string rather than a JSON-formatted string. I'll go ahead and fix that comment.

Reply:
Fixed in dc89107.